sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSourceStrategy.scala

```diff
@@ -573,8 +573,10 @@ object DataSourceStrategy
    * @return a `Some[Filter]` if the input [[Expression]] is convertible, otherwise a `None`.
    */
   protected[sql] def translateFilter(
-      predicate: Expression, supportNestedPredicatePushdown: Boolean): Option[Filter] = {
-    translateFilterWithMapping(predicate, None, supportNestedPredicatePushdown)
+      predicate: Expression,
+      supportNestedPushDown: Boolean,
+      canPartialPushDown: Boolean): Option[Filter] = {
```

Comment:
canPartialPushdown and supportNestedPushDown can have a default value as false, so if some logic does not require these options, no need to define variable before calling function.